peps/pep-0752.rst

```diff
@@ -422,6 +433,145 @@ and ``Y``. If each repository has both packages but one is malicious on ``X``
 and the other is malicious on ``Y`` then the user would be unable to satisfy
 their requirements without encountering a malicious package.
 
+.. _digital-attestations:
+
+Exclusive Reliance on Digital Attestations
+------------------------------------------
+
+The idea [5]_ here would be to solely rely on :pep:`740` attestations to verify
+certain properties of dependencies, such as:
+
+* The package was uploaded by a specific organization or user name
+* The package was uploaded by an owner of a specific domain name
+* The package was uploaded by a user with a specific email address
+
+A general downside to this approach is that implementing this on custom package
+repositories would require a significant amount of work. In the case of PyPI,
+the prerequisite work on `Trusted Publishing`__ and then the
+`PEP 740 implementation`__ itself took the equivalent of a full-time engineer
+one year whose time was paid for by a corporate sponsor. This is unlikely to be
+implemented by enterprise because it's possible to implement reproducible
+builds already using simpler mechanisms, and such attestations are not very
+useful when everything is internally managed.
+
+__ https://blog.pypi.org/posts/2023-04-20-introducing-trusted-publishers/#acknowledgements
+__ https://blog.trailofbits.com/2024/10/01/securing-the-software-supply-chain-with-the-slsa-framework/
+
+The amount of effort wouldn't matter for community projects because they
+wouldn't be able to maintain any infrastructure themselves in most cases.
+
+Another weakness is that a user attempting to perform a simple ``pip install``
+would already be vulnerable to malicious packages. In order to overcome this,
+there would have to be the concept of implicit constraints, all of which would
+impose certain UX or resolver logic upon every tool.
+
+For example, package managers could be changed such that the first time a
+package is installed the user would receive a confirmation prompt displaying
+the attestation details. This would be very confusing and noisy, especially for
+new users, and would be a breaking UX change for existing users. Many methods
+of installation wouldn't work for this scenario such as installing from a
+requirements file where the user would potentially be getting hundreds of
+prompts.
+
+One solution to make this less disruptive for the user would be to manually
+maintain a list of trustworthy details (organization/user names, domain names,
+email addresses, etc.). This could be discoverable by packages providing
+`entry points`__, which corporate environments could install by default. This
+has the major downside of not providing automatic guarantees which would limit
+the usefulness for the average user who is more likely to be affected.
+
+__ https://packaging.python.org/en/latest/specifications/entry-points/
+
+An idea to provide automatic protection would be to have a set of default
+packages that would be installed by default. Each repository could offer a
```

Comment:
"...by default" - with the double "default" maybe the second one could be "...installed without further explicit confirmation." ?

Reply:
Fixed, thanks!